Change ActiveRecord::Migration to ActiveRecord::Migration[7.1]

lenikadali opened the pull request:
- Changed the line `ActiveRecord::Migration to ActiveRecord::Migration[7.1]` so that schema loading could work (it is now required for newer versions of Rails)
- Updated and commented out the load schema task since it fails to find and run the loading of data into the database PostgreSQL container
- Added the initialize_profile method and add it to the `after_create` portion of the User factory because of the `NameError` I encountered:
```bash
NameError: undefined local variable or method `init_profile' for #<FactoryBot::SyntaxRunner:0x00007eaac2e3c560> (NameError)
```

lenikadali committed:
Change ActiveRecord::Migration to ActiveRecord::Migration[7.1]

Changed the line ActiveRecord::Migration to ActiveRecord::Migration[7.1]
so that schema loading could work (it is now required for newer versions
of Rails)

Updated and commented out the load schema task since it fails to find
and run the loading of data into the database PostgreSQL container

Added the initialize_profile method and add it to the `after_create`
portion of the User factory because of the NameError I encountered:

NameError: undefined local variable or method `init_profile'
for #<FactoryBot::SyntaxRunner:0x00007eaac2e3c560> (NameError)

lenikadali commented on backend/spec/factories/users.rb:
```diff
@@ -26,9 +26,22 @@
 
 FactoryBot.define do
+
+  def initialize_profile
+    create_profile!(
+      checkin_reminder: true,
+      onboarding_step_id: "onboarding-personal",
+      most_recent_doses: {},
+      most_recent_conditions_positions: {},
+      most_recent_symptoms_positions: {},
+      most_recent_treatments_positions: {}
+    )
+  end
+
   factory :user do
     sequence(:email) { |number| "user#{number}@example.com" }
     password { "password123" }
     password_confirmation { "password123" }
+    after(:create) { initialize_profile }
```
Modified the `User` factory to have the `init_profile` model from the `User` model and renamed it to `initialize_profile` because of the error below:

```bash
Done.
bin/rails aborted!
NameError: undefined local variable or method `init_profile' for #<FactoryBot::SyntaxRunner:0x00007eaac2e3c560> (NameError)

    after(:create) { init_profile }
                     ^^^^^^^^^^^^
Did you mean?  initialize
/app/spec/factories/users.rb:45:in `block (3 levels) in <top (required)>'
/app/db/seeds/development/users.seeds.rb:4:in `block in evaluate'
/app/db/seeds/development/users.seeds.rb:3:in `each'
/app/db/seeds/development/users.seeds.rb:3:in `evaluate'
Tasks: TOP => db:seed => db:seed:development => db:seed:development:users
(See full trace by running task with --trace)
```

compwron reviewed (commented):
This is looking good!!

haydenrou reviewed (commented):
thank you! Just left a couple of thoughts

lenikadali commented on backend/lib/tasks/app.rake:
```diff
@@ -42,8 +42,9 @@ namespace :app do
+      Rake::Task["db:migrate"].invoke
     end
-    Rake::Task["db:structure:load"].invoke
+    #Rake::Task["db:schema:load"].invoke
```
I wouldn't know since I was using this for development purposes :upside_down_face: 
Maybe the contributors who've deployed Flaredown in production can comment?

compwron commented:
There's some good stuff in here, rebase and we can merge?

lenikadali commented on backend/db/migrate/20170817154145_create_positions.rb:
```diff
@@ -1,10 +1,10 @@
       t.string :location_name, null: false
       t.decimal :latitude, {precision: 10, scale: 7}
-      t.decimal :longitude, {precision: 10, scale: 7}
+      #t.decimal :longitude, {precision: 10, scale: 7}
```
Finally got around to this. Here is the traceback with the above line _**not**_ commented out:

```bash
== 20170817154145 CreatePositions: migrating ==================================
-- create_table(:positions)
bin/rails aborted!
StandardError: An error has occurred, this and all later migrations canceled: (StandardError)

you can't define an already defined column '{:precision=>10, :scale=>7}' on 'positions'.
/app/db/migrate/20170817154145_create_positions.rb:7:in `block in change'
/app/db/migrate/20170817154145_create_positions.rb:3:in `change'
/app/lib/tasks/app.rake:45:in `build_database'
/app/lib/tasks/app.rake:32:in `setup'
/app/lib/tasks/app.rake:4:in `block (2 levels) in <top (required)>'

Caused by:
ArgumentError: you can't define an already defined column '{:precision=>10, :scale=>7}' on 'positions'. (ArgumentError)

              raise ArgumentError, "you can't define an already defined column '#{name}' on '#{@name}'."
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/app/db/migrate/20170817154145_create_positions.rb:7:in `block in change'
/app/db/migrate/20170817154145_create_positions.rb:3:in `change'
/app/lib/tasks/app.rake:45:in `build_database'
/app/lib/tasks/app.rake:32:in `setup'
/app/lib/tasks/app.rake:4:in `block (2 levels) in <top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)

make: *** [Makefile:23: seed] Error 1
```

From what I can tell, it seems that the database for some reason thinks `latitude` and `longtitude` are the same field. If it helps, I am using the Docker setup via `make build` and `make seed`.

lenikadali replied:
Resolved in [17756c8](https://github.com/rubyforgood/Flaredown/pull/788/commits/17756c8eaa0827173e421c39020276cbf1e3085b)

lenikadali committed:
Revert ActiveRecord::Migration class from [7.1] to [5.1]

Reverted changing ActiveRecord::Migration class from [7.1]
to [5.1] (this was done as part of the feature branch rebase
on master)

Also added lint fixes

lenikadali commented:
> lint error in CI
> 
> ```
>  standard: Use Ruby Standard Style (https://github.com/standardrb/standard)
>   db/migrate/20170817154145_create_positions.rb:7:7: Layout/LeadingCommentSpace: Missing space after `#`.
>   lib/tasks/app.rake:47:5: Layout/LeadingCommentSpace: Missing space after `#`.
>   spec/factories/users.rb:29:1: Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.
> standard: Run `standardrb --fix` 
> ```

Resolved in [f00f774](https://github.com/rubyforgood/Flaredown/pull/788/commits/f00f7744e721f67afd82428789473caebe2100ad)

lenikadali committed:
Revert change to fixtures task

Reverted change to fixtures task since going by
https://github.com/rubyforgood/Flaredown/pull/789
the original task (`db:structure:load`) has been
changed to load fixtures instead.

lenikadali commented on backend/lib/tasks/app.rake:
```diff
@@ -44,7 +44,7 @@ namespace :app do
       Rake::Task["db:migrate"].invoke
     end
-    Rake::Task["db:fixtures:load"].invoke
+    #Rake::Task["db:schema:load"].invoke
```
Resolved in [c103d87](https://github.com/rubyforgood/Flaredown/pull/788/commits/c103d876d90994be007cc8a880b64e7b48f1358b)

lenikadali committed:
Remove `initialize_profile`, `after_create` call

Removed the `initialize_profile` method
and the `after_create` call that uses it
since the method may have unforeseen side effects
on test setup.

lenikadali commented on backend/spec/factories/users.rb:
```diff
@@ -26,9 +26,22 @@
     sequence(:email) { |number| "user#{number}@example.com" }
     password { "password123" }
     password_confirmation { "password123" }
+    after(:create) { initialize_profile }
```
Resolved in [235a49a](https://github.com/rubyforgood/Flaredown/pull/788/commits/235a49a58de3b2842e8d86c8300b5889844c0562)

lenikadali committed:
Revert removal of longitude column

lenikadali commented on backend/db/migrate/20170817154145_create_positions.rb:
```diff
@@ -1,10 +1,10 @@
       t.string :location_name, null: false
       t.decimal :latitude, {precision: 10, scale: 7}
-      t.decimal :longitude, {precision: 10, scale: 7}
+      #t.decimal :longitude, {precision: 10, scale: 7}
```
Resolved in [17756c8](https://github.com/rubyforgood/Flaredown/pull/788/commits/17756c8eaa0827173e421c39020276cbf1e3085b)

compwron commented:
this might be fixed, let's see...

lenikadali commented:
> hmm there are 0 files in this PR now?

Yeah it would seem that the changes that were in this PR have been superseded by what is on `master` \o/
So I'm closing this PR as no longer relevant :upside_down_face:
